app/src/main/java/com/github/giommok/softwaredevproject/Account.java

```diff
@@ -52,4 +51,14 @@ public static Account getAccount() {
      * This method updates the username of the current account
      */
     public void synchronizeUsername();
+
+    /**
+     * Check if the username chosen by the user is valid.
+     * @param username inserted by the user.
+     * @return true if and only if the username is valid.
+     */
+    public static Boolean isValid(String username) {

```

Comment:
if(username == null) return false;
        return username.matches("\\w{3,15}");

Could be written as
return (username == null) ? false : username.matches("\\w{3,15}");

Reply:
Good point, it can be written in just one line. However, I will write it like this:
`return username != null && username.matches("\\w{3,15}");`
I've also changed all the methods name in ProfileActivityTest.

Reply:
Yes nice!